### PR TITLE
ADPQVP2018-229: added endpoint for logout and resolved some other iss…

### PR DESCRIPTION
**Pull request by manojmewara-itt**
…ues related to password update

### CODE REVIEW

File: server-src/scripts/setUpDB.sh [hunk 1]
```diff
@@ -14,13 +14,14 @@ mongo kmt_db --eval 'db.articleType.remove({ })'
...
-         "email": "kmtAdmin@mailinator.com",
+         "email": "kmtadmin@mailinator.com",
          "userRole": "admin",
          "firstName": "kmt",
          "lastName": "admin",
          "password": "Admin123",
          "active": true,
-         "dateJoined": new Date()
+         "dateJoined": new Date(),
+		 "session": false
```
Comment (koteswara-v) [inline server-src/scripts/setUpDB.sh]: indentation ?

File: server-src/src/main/java/com/itt/kmt/jwt/JWTFilter.java [hunk 2]
```diff
@@ -67,7 +65,7 @@ protected boolean isAccessAllowed(
             try {
                 executeLogin(request, response);
             } catch (Exception e) {
-                response401(request, response);
+                e.printStackTrace();
```
Comment (koteswara-v) [inline server-src/src/main/java/com/itt/kmt/jwt/JWTFilter.java]: log the exception instead of print

File: server-src/src/main/java/com/itt/kmt/services/UserService.java [hunk 3]
```diff
@@ -128,15 +128,11 @@ public User updateUser(final User user, final String id) {
...
-            if (!user.getFirstName().isEmpty()) {
-                existingUser.setFirstName(user.getFirstName());
-            }
-            if (!user.getLastName().isEmpty()) {
-                existingUser.setLastName(user.getLastName());
-            }
-            if (!user.getUserRole().isEmpty()) {
-                existingUser.setUserRole(user.getUserRole());
-            }
+            existingUser.setFirstName(user.getFirstName());
```
Comment (koteswara-v) [inline server-src/src/main/java/com/itt/kmt/services/UserService.java]: Please revert these changes
We want to check validity of data before writing data to DB

File: server-src/src/main/java/com/itt/utility/Constants.java [hunk 4]
```diff
@@ -13,6 +13,10 @@ private Constants() {
 
         super();
     }
+    
+    /** The Constant USER_LOGOUT_SUCCESS_MSG. */
+    public static final String USER_LOGOUT_SUCCESS_MSG = "User has been logout successfully";
```
Comment (koteswara-v) [inline server-src/src/main/java/com/itt/utility/Constants.java]: Message should be
User has been logged out successfully.

File: server-src/scripts/setUpDB.sh (hunk 1 above)
Comment (manojmewara-itt) [reply server-src/scripts/setUpDB.sh]: it looks fine in source code, showing indentation only on github, you can check once by take pull

File: server-src/src/main/java/com/itt/kmt/services/UserService.java (hunk 3 above)
Comment (manojmewara-itt) [reply server-src/src/main/java/com/itt/kmt/services/UserService.java]: We already added the validator for user add/update that will take care of validations of it, no need to recheck here
